packages/core-plugin-api/src/analytics/useAnalytics.tsx

```diff
@@ -38,22 +38,32 @@ function useAnalyticsApi(): AnalyticsApi {
  * @public
  */
 export function useAnalytics(): AnalyticsTracker {
-  const trackerRef = useRef<Tracker | null>(null);
   const context = useAnalyticsContext();
-  // Our goal is to make this API truly optional for any/all consuming code
-  // (including tests). This hook runs last to ensure hook order is, as much as
-  // possible, maintained.
   const analyticsApi = useAnalyticsApi();
 
-  function getTracker(): Tracker {
-    if (trackerRef.current === null) {
-      trackerRef.current = new Tracker(analyticsApi);
-    }
-    return trackerRef.current;
-  }
-
-  const tracker = getTracker();
-  tracker.setContext(context);
-
-  return tracker;
+  return useMemo(() => {
```

Comment:
Think the `e2e` test fails are happening on every branch right now, but the unit test fails in this branch seem to be unique to this branch, and are probably related to the changes in here.